src/Build/Evaluation/Context/EvaluationContext.cs

```diff
@@ -58,6 +58,14 @@ public static EvaluationContext Create(SharingPolicy policy)
             return context;
         }
 
+        private EvaluationContext CreateUsedIsolatedContext()
+        {
+            var context = Create(SharingPolicy.Isolated);
+            context._used = 1;
```

Comment:
nit: this may not be needed as it is already handled by the Interlocked.CompareExchange below.
- if previous value was 0, it gets switched to 1 (and method doesn't even get called)
- if previous value was 1, it stays as 1 - that's also when CreatedIsolatedContext gets called

Reply:
Nevermind, I didn't realize that is on a new object.